tests/api/ai_explain.rs

```diff
@@ -0,0 +1,110 @@
+use crate::helpers::app::test_app_with_login;
+use crate::helpers::db::{get_pool, reset};
+use crate::helpers::wait_for_stubr;
+use actix_web::test;
+use anyhow::Error;
+use diesel::{QueryDsl, RunQueryDsl};
+use hmac::Mac;
+use rumba::ai::explain::{hash_highlighted, ExplainRequest, HmacSha256, AI_EXPLAIN_VERSION};
+use rumba::db::ai::{add_explain_answer, ExplainFeedback, FeedbackTyp};
+use rumba::db::model::{AIExplainCacheInsert, AIExplainCacheQuery};
+use rumba::db::schema::ai_explain_cache;
+use rumba::settings::SETTINGS;
+
+const JS_SAMPLE: &str = "const foo = 1;";
+
+fn sign(sample: &str) -> Result<Vec<u8>, Error> {
+    let mut mac = HmacSha256::new_from_slice(
+        &SETTINGS
+            .ai
+            .as_ref()
+            .map(|ai| ai.explain_sign_key)
+            .expect("missing sign_key"),
+    )?;
+
+    mac.update(sample.as_bytes());
+
+    Ok(mac.finalize().into_bytes().to_vec())
+}
+
+fn add_explain_cache() -> Result<(), Error> {
+    let insert = AIExplainCacheInsert {
+        language: Some("js".to_owned()),
+        signature: sign(JS_SAMPLE)?,
+        highlighted_hash: hash_highlighted(JS_SAMPLE),
+        explanation: Some("Explain this!".to_owned()),
+        version: AI_EXPLAIN_VERSION,
+    };
+    let pool = get_pool();
+    let mut conn = pool.get()?;
+    add_explain_answer(&mut conn, &insert)?;
+    Ok(())
+}
+
+#[actix_rt::test]
+#[stubr::mock(port = 4321)]
+async fn test_explain() -> Result<(), Error> {
+    let pool = reset()?;
+    add_explain_cache()?;
+    wait_for_stubr().await?;
+    let app = test_app_with_login(&pool).await.unwrap();
+    let service = test::init_service(app).await;
+    let request = test::TestRequest::post()
+        .uri("/api/v1/plus/ai/explain")
+        .set_json(ExplainRequest {
+            language: None,
```

Comment:
It's intentional that we return the cached js response, despite the language being different?

Reply:
yes

Reply:
But maybe I should solve for that 😭